### PR TITLE
filter certificates from db and implement identity helper

### PR DESCRIPTION
**Pull request by Dinek007**

### CODE REVIEW

File: src/renderer/store/certificates/certificates.saga.ts [hunk 1]
```diff
@@ -2,19 +2,38 @@ import { call, apply, all, takeEvery, put, select } from 'typed-redux-saga'
...
 import directMessagesSelectors from '../selectors/directMessages'
 
+const filterCertificates = (certificates: string[]): string[] => {
+  return certificates.filter((hasCertyficate) => {
+    const certFieldsArray = Object.keys(CertFieldsTypes)
+    const parsedCert = parseCertificate(hasCertyficate)
+
+    let isValid = true
+
+    for (let i = 0; i < certFieldsArray.length; i++) {
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:20]: Change to 

for (const field of Object.keys(CertFieldsTypes))

Better than enumerating with index

**Review by kowalski — CHANGES_REQUESTED** [review]

**Comment by holmesworcester** [comment]
So we want to add the OrbitDB identity to the CSR, correct?

File: src/renderer/store/certificates/certificates.saga.ts [hunk 2]
```diff
@@ -2,19 +2,36 @@ import { call, apply, all, takeEvery, put, select } from 'typed-redux-saga'
...
+  })
+}
+
 export function* responseGetCertificates(
   action: PayloadAction<ReturnType<typeof certificatesActions.responseGetCertificates>['payload']>
 ): Generator {
   const certificates = action.payload
-  yield* put(certificatesActions.setUsersCertificates(certificates.certificates))
+
+  const filteredCertificates = yield* call(filterCertificates, certificates.certificates)
```
Comment (kowalski) [inline src/renderer/store/certificates/certificates.saga.ts:33]: No need for yield call, there is no side effect in this function. 
you can call filterCertificates directly

File: src/renderer/store/selectors/contacts.ts [hunk 3]
```diff
@@ -208,11 +208,12 @@ const usersCertificateMapping = createSelector(
       if (current !== null && current) {
         parsedCerficated = extractPubKeyString(current)
         certObject = loadCertificate(current)
-        nickname = certObject.subject.typesAndValues[0].value.valueBlock.value
-        onionAddress = certObject.subject.typesAndValues[1].value.valueBlock.value
-        peerId = certObject.subject.typesAndValues[2].value.valueBlock.value
-        const dmPubKeyArray = certObject.subject.typesAndValues[3]?.value.valueBlock.valueHex
-        dmPubKey = arrayBufferToHexString(dmPubKeyArray)
+        if (certObject.subject.typesAndValues.length === 4) {
```
Comment (kowalski) [inline src/renderer/store/selectors/contacts.ts:211]: Instead of checking length of attributes just check that you have all the values

**Review by kowalski — APPROVED** [review]

**Pull request merged**